src/main/java/io/vertx/core/net/TrafficShapingOptions.java

```diff
@@ -139,14 +139,14 @@ public TrafficShapingOptions setMaxDelayToWaitUnit(TimeUnit maxDelayToWaitTimeUn
   }
 
   /**
-   * Set the delay between two computations of performances for channels or 0 if no stats are to be computed
+   * Set the delay between two computations of performances for channels
    *
    * @param checkIntervalForStats delay between two computations of performances
    * @return a reference to this, so the API can be used fluently
    */
   public TrafficShapingOptions setCheckIntervalForStats(long checkIntervalForStats) {
     this.checkIntervalForStats = checkIntervalForStats;
-    ObjectUtil.checkPositive(this.checkIntervalForStats, "checkIntervalForStats");
+    ObjectUtil.checkPositiveOrZero(this.checkIntervalForStats, "checkIntervalForStats");
```

Comment:
looking at the netty code, nothing prevents us from having a negative value for this field. Any reason we are restricting it to zero or positive?

```
    public void configure(long newCheckInterval) {
        long newInterval = newCheckInterval / 10 * 10;
        if (checkInterval.getAndSet(newInterval) != newInterval) {
            if (newInterval <= 0) {
                stop();
                // No more active monitoring
                lastTime.set(milliSecondFromNano());
            } else {
                // Restart
                stop();
                start();
            }
        }
    }
```

Reply:
Since negative values and 0 have the same effect. I thought we can use 0 for stop monitoring and check for disallow negative values to avoid confusion

Reply:
makes sense. Previously we were disallowing disabling it, and this addresses that issue.